packages/oauth-client/src/CognitoClient.ts

```diff
@@ -0,0 +1,145 @@
+import got, { HTTPError, Response } from 'got';
+
+import { URL } from 'url';
+
+import { OAuthClient } from './OAuthClient';
+import { CognitoError } from './CognitoError';
+
+type CognitoErrorResponse = Response<{
+  error: string,
+  error_description: string,
+}>;
+
+const validateUrl = (urlString: string) => {
+  // eslint-disable-next-line no-new
+  new URL(urlString);
+};
+
+const httpErrorToCognitoError = (httpError: HTTPError) => {
+  const response = <CognitoErrorResponse>httpError.response;
+
+  if (response.body && response.body.error) {
+    return new CognitoError(
+      response.body.error,
+      response.body.error_description
+    );
+  }
+
+  return new CognitoError(
+    'UnexpectedResponse',
+    `Unexpected response: ${httpError.response.body}`
+  );
+};
+
+/**
+ * A client for the Cognito API
+ */
+export class CognitoClient extends OAuthClient {
+  readonly clientId: string;
+  readonly clientPassword: string;
+  readonly cognitoLoginUrl: string;
+  readonly redirectUri: string;
+
+  /**
+   * @param {Object} params
+   * @param {string} params.clientId - see example
+   * @param {string} params.clientPassword - see example
+   * @param {string} params.cognitoLoginUrl - see example
+   * @param {string} params.redirectUri - see example
+   *
+   * @example
+   *
+   * const oAuth2Provider = new CognitoClient({
+   *   clientId: 'my-client-id',
+   *   clientPassword: 'my-client-password',
+   *   cognitoLoginUrl: 'https://earthdata.login.nasa.gov',
+   *   redirectUri: 'http://my-api.com'
+   * });
+   */
+
+  constructor(
+    params: {
+      clientId: string,
+      clientPassword: string,
+      cognitoLoginUrl: string,
+      redirectUri: string
+    }
+  ) {
+    if (!params.clientId) throw new TypeError('clientId is required');
```

Comment:
Yup, that's much cleaner. The `EarthdataLoginClient` still includes the original assignments because we have that annoying `earthdataLoginUrl` to preserve the original interface. I did remove the validation though. The CognitoClient is much cleaner:

https://github.com/nasa/cumulus/pull/2209/commits/f23d283114a5f63c54b14a6b644ab76b5889f103